### ISSUE TITLE
Add functionality to `Ecto.SoftDelete.Repo` to exclude soft deleted records from Repo queries by default

### ISSUE THREAD

**Issue by Dacello** [revelrylabs/ecto_soft_delete#78]
Soft deleted records are included by default in all ecto queries. This is less than ideal, particularly when preloading associations. Would be really nice if we could just assume that any time we do a `Repo.all/get/one` as well as an ecto preload, soft deleted records were excluded by default. We could potentially bake this functionality into `Ecto.SoftDelete.Repo` so that it overrides the standard `Ecto.Repo` functionality for these functions.

**Comment by Dacello** [comment]
@bryanjos FYI I have been made aware that with newer versions of ecto, we can add where clauses to schema associations. Maybe we could utilize this to add things like `has_many_undeleted` association methods to this?

**Comment by NOrzechowski** [comment]
Notes: 

https://hexdocs.pm/ecto/Ecto.Repo.html#c:prepare_query/3

```
def prepare_query(_operation, query, opts) do
    if opts[:admin] do
      {query, opts}
    else
      query = from(x in query, where: is_nil(x.deleted_at))
      {query, opts}
    end
  end
```

The problem with that ^ is not all tables have `deleted_at`.

**Comment by Dacello** [comment]
Fixed in #86, published as new version in tag `v2.0.0`
